src/main/java/ch/njol/skript/sections/EffSecOpenInventory.java

```diff
@@ -0,0 +1,283 @@
+	"When using as a section. The section allows for modification of the inventory via the event-inventory."
+})
+@Examples({
+	"show crafting table to player #unmodifiable, use open instead to allow for recipes to work",
+	"open a crafting table to the player",
+	"open a loom to the player",
+	"open the player's inventory for the player",
+	"",
+	"show chest inventory to player:",
+	"\tset slot 1 of event-inventory to stone named \"example\"",
+	"\topen event-inventory to all players"
+})
+@Since("2.0, 2.1.1 (closing), 2.2-Fixes-V10 (anvil), INSERT VERSION (enchanting, cartography, grindstone, loom) & section support")
+public class EffSecOpenInventory extends EffectSection {
+
+	private static enum OpenableInventorySyntax {
+
+		ANVIL("anvil"),
+		CARTOGRAPHY("cartography [table]", Skript.methodExists(HumanEntity.class, "openCartographyTable", Location.class, boolean.class),
+				"Opening a cartography table inventory requires Paper."),
+		ENCHANTING("enchant(ment|ing) [table]", new Version(1, 14)),
+		GRINDSTONE("grindstone", Skript.methodExists(HumanEntity.class, "openGrindstone", Location.class, boolean.class),
+				"Opening a grindstone inventory requires Paper."),
+		LOOM("loom", Skript.methodExists(HumanEntity.class, "openLoom", Location.class, boolean.class),
+				"Opening a loom inventory requires Paper."),
+		SMITHING("smithing [table]", Skript.methodExists(HumanEntity.class, "openSmithingTable", Location.class, boolean.class),
+				"Opening a smithing table inventory requires Paper."),
+		STONECUTTER("stone[ ]cutter", Skript.methodExists(HumanEntity.class, "openSmithingTable", Location.class, boolean.class),
+				"Opening a stone cutter inventory requires Paper."),
+		WORKBENCH("(crafting [table]|workbench)");
+
+		private @Nullable String methodError;
+		private @Nullable Version version;
+		private final String property;
+		private boolean methodExists = true;
+
+		OpenableInventorySyntax(String property) {
+			this.property = property;
+		}
+
+		OpenableInventorySyntax(String property, Version version) {
+			this.property = property;
+			this.version = version;
+		}
+
+		OpenableInventorySyntax(String property, boolean methodExists, String methodError) {
+			this.methodExists = methodExists;
+			this.methodError = methodError;
+			this.property = property;
+		}
+
+		private String getFormatted() {
+			return this.toString().toLowerCase(Locale.ENGLISH) + ":" + property;
+		}
+
+		@Nullable
+		private Version getVersion() {
+			return version;
+		}
+
+		private boolean doesMethodExist() {
+			return methodExists;
+		}
+
+		@Nullable
+		private String getMethodError() {
+			return methodError;
+		}
+
+		private static String construct() {
+			StringBuilder builder = new StringBuilder("(");
+			OpenableInventorySyntax[] values = OpenableInventorySyntax.values();
+			for (int i = 0; i < values.length; i++ ) {
+				builder.append(values[i].getFormatted());
+				if (i + 1 < values.length)
+					builder.append("|");
```

Comment:
We require `__ inventory` for all normal inventories, shouldn't these require it too?
eg for barrels you have to `open barrel inventory` but for anvil you can just do `open anvil` right now

Reply:
This would break the existing syntax.